lib/gst/player/gstplayer-media-info.h

```diff
@@ -0,0 +1,119 @@
+  (G_TYPE_CHECK_INSTANCE_TYPE((obj),GST_TYPE_PLAYER_VIDEO_INFO))
+
+typedef struct _GstPlayerVideoInfo GstPlayerVideoInfo;
+typedef GObjectClass GstPlayerVideoInfoClass;
+GType gst_player_video_info_get_type (void);
+
+guint gst_player_video_info_get_bitrate(const GstPlayerVideoInfo* info);
+guint gst_player_video_info_get_max_bitrate(const GstPlayerVideoInfo* info);
+guint gst_player_video_info_get_width(const GstPlayerVideoInfo* info);
+guint gst_player_video_info_get_height(const GstPlayerVideoInfo* info);
+guint gst_player_video_info_get_framerate_num(const GstPlayerVideoInfo* info);
+guint gst_player_video_info_get_framerate_denom(const GstPlayerVideoInfo* info);
+
+#define GST_TYPE_PLAYER_AUDIO_INFO \
+  (gst_player_audio_info_get_type ())
+#define GST_PLAYER_AUDIO_INFO(obj) \
+  (G_TYPE_CHECK_INSTANCE_CAST((obj),GST_TYPE_PLAYER_AUDIO_INFO, GstPlayerAudioInfo))
+#define GST_IS_PLAYER_AUDIO_INFO(obj) \
+  (G_TYPE_CHECK_INSTANCE_TYPE((obj),GST_TYPE_PLAYER_AUDIO_INFO))
+
+typedef struct _GstPlayerAudioInfo GstPlayerAudioInfo;
+typedef GObjectClass GstPlayerAudioInfoClass;
+GType gst_player_audio_info_get_type (void);
+
+guint gst_player_audio_info_get_channels(const GstPlayerAudioInfo* info);
+guint gst_player_audio_info_get_sample_rate(const GstPlayerAudioInfo* info);
+guint gst_player_audio_info_get_depth(const GstPlayerAudioInfo* info);
+guint gst_player_audio_info_get_bitrate(const GstPlayerAudioInfo* info);
+guint gst_player_audio_info_get_max_bitrate(const GstPlayerAudioInfo* info);
+const gchar * gst_player_audio_info_get_language(const GstPlayerAudioInfo* info);
+
+#define GST_TYPE_PLAYER_SUBTITLE_INFO \
+  (gst_player_subtitle_info_get_type ())
+#define GST_PLAYER_SUBTITLE_INFO(obj) \
+  (G_TYPE_CHECK_INSTANCE_CAST((obj),GST_TYPE_PLAYER_SUBTITLE_INFO, GstPlayerSubtitleInfo))
+#define GST_IS_PLAYER_SUBTITLE_INFO(obj) \
+  (G_TYPE_CHECK_INSTANCE_TYPE((obj),GST_TYPE_PLAYER_SUBTITLE_INFO))
+
+typedef struct _GstPlayerSubtitleInfo GstPlayerSubtitleInfo;
+typedef GObjectClass GstPlayerSubtitleInfoClass;
+GType gst_player_subtitle_info_get_type (void);
+
+const gchar * gst_player_subtitle_info_get_language(const GstPlayerSubtitleInfo* info);
+
+#define GST_TYPE_PLAYER_MEDIA_INFO \
+  (gst_player_media_info_get_type())
+#define GST_PLAYER_MEDIA_INFO(obj) \
+  (G_TYPE_CHECK_INSTANCE_CAST((obj),GST_TYPE_PLAYER_MEDIA_INFO,GstPlayerMediaInfo))
+#define GST_PLAYER_MEDIA_INFO_CLASS(klass) \
+  (G_TYPE_CHECK_CLASS_CAST((klass),GST_TYPE_PLAYER_MEDIA_INFO,GstPlayerMediaInfoClass))
+#define GST_IS_PLAYER_MEDIA_INFO(obj) \
+  (G_TYPE_CHECK_INSTANCE_TYPE((obj),GST_TYPE_PLAYER_MEDIA_INFO))
+#define GST_IS_PLAYER_MEDIA_INFO_CLASS(klass) \
+  (G_TYPE_CHECK_CLASS_TYPE((klass),GST_TYPE_PLAYER_MEDIA_INFO))
+
+typedef struct _GstPlayerMediaInfo GstPlayerMediaInfo;
+typedef GObjectClass GstPlayerMediaInfoClass;
+GType gst_player_media_info_get_type (void);
+
+GstPlayerMediaInfo* gst_player_media_info_new (const gchar *uri);
```

Comment:
The _new() function should be private, only GstPlayer is ever going to call it and not the application